### ISSUE TITLE
Compact and WithoutEmpty are the same thing

### ISSUE THREAD

**Issue by elliot-u410** [samber/lo#261]
Should one be deprecated in favor of the other?

**Comment by samber** [comment]
Indeed, thanks for the feedback.

- `WithoutEmpty` does not exist in `Lodash`
- `lo` has a `Without` helper

I would deprecate `Compact`, which I consider less self-explaining, even if `Lodash` chose to name it `Compact`. Also, I like the idea of having `Without` + `WithoutEmpty` variations.

WDYT ?

**Comment by elliot-u410** [comment]
I agree `Compact` is less descriptive. Deprecation sounds like the best option.

**Comment by elliot-u410** [comment]
Note that the *most* descriptive option is probably something like `lo.Filter(xs, lo.IsNotEmpty)` which is not much longer than `lo.WithoutEmpty(xs)`.
